spring-pulsar/src/test/java/org/springframework/pulsar/core/reactive/ReactivePulsarTemplateTests.java

```diff
@@ -164,32 +177,36 @@ void sendMessageTest(String testName, SendTestArgs testArgs) throws Exception {
 	private static Stream<Arguments> sendMessageTestProvider() {
 		return Stream.of(arguments("sendReactiveMessageToDefaultTopic", SendTestArgs.useSpecificTopic(false)),
 				arguments("sendReactiveMessageToDefaultTopicWithSimpleApi",
-						SendTestArgs.useSpecificTopic(false).useSimpleApi(true)),
+						SendTestArgs.useSpecificTopic(false).useSimpleApi()),
+				arguments("sendReactiveMessageToDefaultTopicWithSimpleApiAndTemplateSchema",
+						SendTestArgs.useSpecificTopic(false).useSimpleApi().useTemplateSchema()),
 				arguments("sendReactiveMessageToDefaultTopicWithRouter",
-						SendTestArgs.useSpecificTopic(false).useCustomRouter(true)),
+						SendTestArgs.useSpecificTopic(false).useCustomRouter()),
 				arguments("sendReactiveMessageToDefaultTopicWithMessageCustomizer",
-						SendTestArgs.useSpecificTopic(false).useMessageCustomizer(true)),
+						SendTestArgs.useSpecificTopic(false).useMessageCustomizer()),
 				arguments("sendReactiveMessageToDefaultTopicWithProducerCustomizer",
-						SendTestArgs.useSpecificTopic(false).useSenderCustomizer(true)),
+						SendTestArgs.useSpecificTopic(false).useSenderCustomizer()),
 				arguments("sendReactiveMessageToDefaultTopicWithAllOptions",
-						SendTestArgs.useSpecificTopic(false).useCustomRouter(true).useMessageCustomizer(true)
-								.useSenderCustomizer(true)),
+						SendTestArgs.useSpecificTopic(false).useCustomRouter().useMessageCustomizer()
+								.useSenderCustomizer()),
 				arguments("sendReactiveMessageToSpecificTopic", SendTestArgs.useSpecificTopic(true)),
 				arguments("sendReactiveMessageToSpecificTopicWithSimpleApi",
-						SendTestArgs.useSpecificTopic(true).useSimpleApi(true)),
+						SendTestArgs.useSpecificTopic(true).useSimpleApi()),
+				arguments("sendReactiveMessageToSpecificTopicWithSimpleApiAndTemplateSchema",
+						SendTestArgs.useSpecificTopic(true).useSimpleApi().useTemplateSchema()),
 				arguments("sendReactiveMessageToSpecificTopicWithRouter",
-						SendTestArgs.useSpecificTopic(true).useCustomRouter(true)),
+						SendTestArgs.useSpecificTopic(true).useCustomRouter()),
 				arguments("sendReactiveMessageToSpecificTopicWithMessageCustomizer",
-						SendTestArgs.useSpecificTopic(true).useMessageCustomizer(true)),
+						SendTestArgs.useSpecificTopic(true).useMessageCustomizer()),
 				arguments("sendReactiveMessageToSpecificTopicWithProducerCustomizer",
-						SendTestArgs.useSpecificTopic(true).useSenderCustomizer(true)),
+						SendTestArgs.useSpecificTopic(true).useSenderCustomizer()),
 				arguments("sendReactiveMessageToSpecificTopicWithAllOptions", SendTestArgs.useSpecificTopic(true)
-						.useCustomRouter(true).useMessageCustomizer(true).useSenderCustomizer(true)));
+						.useCustomRouter().useMessageCustomizer().useSenderCustomizer()));
 	}
 
 	static final class SendTestArgs {
 
-		private boolean useSpecificTopic;
+		private final boolean useSpecificTopic;
```

Comment:
~Does this have to be final? The rule we use when to use final is when it is an immutable object w/ constructor injection.~ Ok, this is the immutable constructed portion of it - good eye.